### PR TITLE
Added EfficientNet models

### LINKED ISSUE #520: Include EfficientNet models
This should be ported from the core Keras API: keras.applications following the example set in: https://github.com/keras-team/keras-cv/pull/476

### PR DESCRIPTION
**Pull request by ayyucedemirbas**
# What does this PR do?

Fixes #520 
Added EfficientNet models

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case. https://github.com/keras-team/keras-cv/issues/520
- [ ] Did you write any new necessary tests?

@LukeWood @tanzhenyu

### CODE REVIEW

File: keras_cv/models/efficientnet.py [hunk 1]
```diff
@@ -0,0 +1,737 @@
...
+
+  Note: each Keras Application expects a specific kind of input preprocessing.
+  For EfficientNet, input preprocessing is included as part of the model
+  (as a `Rescaling` layer), and thus
+  `tf.keras.applications.efficientnet.preprocess_input` is actually a
+  pass-through function. EfficientNet models expect their inputs to be float
+  tensors of pixels with values in the [0-255] range.
+
+  Args:
+    include_top: Whether to include the fully-connected
```
Comment (LukeWood) [inline keras_cv/models/efficientnet.py:155]: docstring should match the new API - please update arguments

File: keras_cv/models/efficientnet.py [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+        layer at the top of the network. Defaults to True.
+    weights: One of `None` (random initialization),
+          'imagenet' (pre-training on ImageNet),
```
Comment (LukeWood) [inline keras_cv/models/efficientnet.py:158]: "imagenet" only will work if we have the right normalization technique included

File: keras_cv/models/efficientnet.py [hunk 3]
```diff
@@ -0,0 +1,737 @@
...
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    # Determine proper input shape
+    input_shape = imagenet_utils.obtain_input_shape(
```
Comment (LukeWood) [inline keras_cv/models/efficientnet.py:281]: please update the code patterns to match those in keras_cv/models/

**Review by LukeWood — CHANGES_REQUESTED** [review]

**Comment by ayyucedemirbas** [comment]
Thank you so much! All done!

**Comment by LukeWood** [comment]
Closed in lieu of https://github.com/keras-team/keras-cv/pull/740
